crates/settlement-clients/ethereum/src/lib.rs

```diff
@@ -113,3 +175,104 @@ impl SettlementClient for EthereumSettlementClient {
         Ok(block_number.try_into()?)
     }
 }
+
+/// To prepare the sidecar for EIP 4844 transaction
+async fn prepare_sidecar(
+    state_diff: &[Vec<u8>],
+    trusted_setup: &KzgSettings,
+) -> Result<(Vec<FixedBytes<131072>>, Vec<FixedBytes<48>>, Vec<FixedBytes<48>>)> {
+    let mut sidecar_blobs = vec![];
+    let mut sidecar_commitments = vec![];
+    let mut sidecar_proofs = vec![];
+
+    for blob_data in state_diff {
+        let fixed_size_blob: [u8; BYTES_PER_BLOB] = blob_data.as_slice().try_into()?;
```

Comment:
we're using `BYTES_PER_BLOB` in multiple places which comes from `alloy`. this isn't the correct design. we have implemented `max_bytes_per_blob` on da clients, we should actually use that over here

Reply:
same as above we need to refactor a lot of code for this to happen. Will discuss this tomorrow in office.

Reply:
hmm, why can't we just get the da client from config and call max bytes per blob? can discuss in office also :)